pulp_container/app/viewsets.py

```diff
@@ -1155,17 +1159,23 @@ class ContainerDistributionViewSet(DistributionViewSet, RolesMixin):
                 "principal": "authenticated",
                 "effect": "allow",
             },
+            # is this redundant?
+            # {
+            #    "action": ["create"],
+            #    "principal": "authenticated",
+            #    "effect": "allow",
+            #    "condition": "has_namespace_model_perms",
+            # },
             {
                 "action": ["create"],
                 "principal": "authenticated",
                 "effect": "allow",
-                "condition": "has_namespace_model_perms",
-            },
-            {
-                "action": ["create"],
-                "principal": "authenticated",
-                "effect": "allow",
-                "condition": "has_namespace_perms:container.add_containerdistribution",
+                "condition": [
+                    "has_namespace_perms:container.add_containerdistribution",
+                    "has_repo_or_repo_ver_param_model_or_obj_perms:container.view_containerrepository",
+                    "has_repo_or_repo_ver_param_model_or_obj_perms:container.view_containerpushrepository",
```

Comment:
this is not correct, container push repos access management is deferred to its distributions/namespaces